### DISCUSSION TITLE
Export to Parquet Functionality is very slow.

### DISCUSSION THREAD

**Discussion by akash-goel** [duckdb/duckdb#6306] [General]
Hi Team,

i am trying to export Duck db database table to Parquet format , Database table contains 64 million records when i am tried to export the data using this query "con.execute("COPY test_df TO 'result-snappy.parquet' (FORMAT 'PARQUET', CODEC 'SNAPPY')")" where test_df is the duck db table , Than its taking approax 20 minutes to create the Parquet on 128 RAM Machine 32 CPU ( No other process running) , Can you please suggest is this expected performance of Export functionality or is there any better way to export the data to Parquet.

Regards,
Akash

**Comment by Mytherin** [comment]
You can enable parallel Parquet export by disabling the preservation of insertion order (`SET preserve_insertion_order=false`) or by toggling the `PER_THREAD_OUTPUT` flag (see #5412).
